azure-pipeline-templates/e2e-tests-block-cache-data-integrity.yml

```
@@ -0,0 +1,206 @@
+      ls -l ${{ parameters.mount_dir }}/datafiles_*
+    displayName: 'Generate data with File-Cache'
+
+  - script: |
+      md5sum ${{ parameters.mount_dir }}/datafiles_* > $(WORK_DIR)/md5sum_original_files.txt
+      md5sum ${{ parameters.mount_dir }}/mixedfiles_* >> $(WORK_DIR)/md5sum_original_files.txt
+    displayName: 'Generate md5Sum with File-Cache' 
+
+  - script: |
+      echo "----------------------------------------------"
+      ls -l ${{ parameters.mount_dir }}
+    displayName: 'Print contents of File-Cache'
+
+  - script: |
+      $(WORK_DIR)/blobfuse2 unmount all
+    displayName: 'Unmount RW mount'
+
+  - script: |
+      cd $(WORK_DIR)
+      $(WORK_DIR)/blobfuse2 gen-test-config --config-file=$(WORK_DIR)/testdata/config/azure_key_bc.yaml --container-name=${{ parameters.container }} --temp-path=${{ parameters.temp_dir }} --output-file=${{ parameters.config_file }}
+      sed -i 's/block-size-mb: [0-9]*/block-size-mb: ${{ parameters.block_size_mb }}/' ${{ parameters.config_file }}
+    displayName: 'Create Config File for RO mount'
+    env:
+      NIGHTLY_STO_ACC_NAME: ${{ parameters.account_name }}
+      NIGHTLY_STO_ACC_KEY: ${{ parameters.account_key }}
+      ACCOUNT_TYPE: ${{ parameters.account_type }}
+      ACCOUNT_ENDPOINT: ${{ parameters.account_endpoint }}
+      VERBOSE_LOG: ${{ parameters.verbose_log }}
+    continueOnError: false
+
+  - script:
+      cat ${{ parameters.config_file }}
+    displayName: 'Print config file'
+
+  - template: 'mount.yml'
+    parameters:
+      working_dir: $(WORK_DIR)
+      mount_dir: ${{ parameters.mount_dir }}
+      temp_dir: ${{ parameters.temp_dir }}
+      prefix: ${{ parameters.idstring }}
+      ro_mount: true
+      mountStep: 
+        script: |
+          $(WORK_DIR)/blobfuse2 mount ${{ parameters.mount_dir }} --config-file=${{ parameters.config_file }} --default-working-dir=$(WORK_DIR) -o ro
+
+  - script: |
+      echo "----------------------------------------------"
+      ls -l ${{ parameters.mount_dir }}/mixedfiles*
+      ls -l ${{ parameters.mount_dir }}/mixedfiles*
+    displayName: 'Print contents of Block-Cache'
+
+  - script: |
+      md5sum ${{ parameters.mount_dir }}/datafiles_* > $(WORK_DIR)/md5sum_block_cache.txt
+      md5sum ${{ parameters.mount_dir }}/mixedfiles_* >> $(WORK_DIR)/md5sum_block_cache.txt
+    displayName: 'Generate md5Sum with Block-Cache'
+
+  - script: |
+      $(WORK_DIR)/blobfuse2 unmount all
+    displayName: 'Unmount RW mount'
+
+  - script: |
+      echo "----------------------------------------------"
+      cat $(WORK_DIR)/md5sum_original_files.txt
+      cat $(WORK_DIR)/md5sum_original_files.txt | cut -d " " -f1 > $(WORK_DIR)/temp.txt && mv temp.txt $(WORK_DIR)/md5sum_original_files.txt
+      echo "----------------------------------------------"
+      cat $(WORK_DIR)/md5sum_block_cache.txt
+      cat $(WORK_DIR)/md5sum_block_cache.txt | cut -d " " -f1 > $(WORK_DIR)/temp.txt && mv temp.txt $(WORK_DIR)/md5sum_block_cache.txt
```

Saw this error in ADO, `mv: cannot stat 'temp.txt': No such file or directory`
```suggestion
      cat $(WORK_DIR)/md5sum_original_files.txt | cut -d " " -f1 > $(WORK_DIR)/temp.txt && mv $(WORK_DIR)/temp.txt $(WORK_DIR)/md5sum_original_files.txt
      echo "----------------------------------------------"
      cat $(WORK_DIR)/md5sum_block_cache.txt
      cat $(WORK_DIR)/md5sum_block_cache.txt | cut -d " " -f1 > $(WORK_DIR)/temp.txt && mv $(WORK_DIR)/temp.txt $(WORK_DIR)/md5sum_block_cache.txt
```